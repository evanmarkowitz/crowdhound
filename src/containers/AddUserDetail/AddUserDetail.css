.add-user-detail {
  padding-top: 60px;
  height: calc(100vh - 60px);
}

.add-user-detail-form {
  margin-top: 20px;
  margin-left: auto;
  margin-right: auto;
  width: 345px;
  display: flex;
  flex-direction: column;
}

.add-user-detail-title {
  color: #f08080;
  font-size: 36px;
  text-align: center;
  margin-bottom: 25px;
}

.label {
  color: rgb(15, 141, 170);
  padding-right: 5px;
  font-size: 22px;
  margin-top: 30px;
}

.input {
  outline: none;
  width: 300px;
  height: 30px;
  border-radius: 5px;
  font-size: 16px;
  border: 1px solid rgb(29, 187, 223);
  padding-left: 7px;
}

#add-user-detail-btn {
  margin-top: 40px;
  cursor: pointer;
  background: rgba(29, 188, 223, 0.253);
}

#user-detail-description-input {
  resize: none;
  height: 200px;
  padding: 15px;
}

#add-user-detail-btn:hover {
  background: rgb(29, 187, 223);
  color: #fff;
}

#add-user-detail-btn:active {
  background: rgb(21, 164, 196);
  color: #fff;
}

.address-container {
  width: 300px;
  height: 150px;
}

.address-input {
  width: 100%;
  height: 30px;
  border-radius: 3px;
  border: 1px solid rgb(29, 187, 223);
  margin-top: 5px;
  outline: none;
  font-size: 14px;
  padding-left: 7px;
}





