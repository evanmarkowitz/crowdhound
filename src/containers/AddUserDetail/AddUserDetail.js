import React, { Component } from 'react';
import { connect } from 'react-redux';
import './AddUserDetail.css';
import { addUserDetailsQuery, createPhoto } from '../../api/apiCallsNew'
import ReactFileReader from 'react-file-reader';
import { Link } from 'react-router-dom'


export class AddUserDetail extends Component {

  constructor(props) {
    super(props)
    this.state = {
      description: '',
      street: '',
      city: '',
      st: '',
      zip: '',
      photo: '', 
    }
  }
  
  handleChange = (e) => {
    this.setState({[e.target.name]: e.target.value})
  }


  handleFiles = (files) => {
    this.sendPhoto(files.base64)
  }

  sendUser = async () => {
    const {firstName, lastName} = this.props
    const {description, street, city, st, zip } = this.state
    let opts = {
      method: 'POST',
      headers: { "Content-Type": "application/json"},
      body: JSON.stringify({query: addUserDetailsQuery(
        firstName, lastName, description, `"${street}"`, city, `"${st}"`, `"${zip}"`
      )})
    }
    try{
      let response = await fetch(`https://staging-crowdhound-be.herokuapp.com/graphql?token=${this.props.token}`, opts)
      let parsedResponse = await response.json()
      await console.log(parsedResponse)

    } catch(error) {
      this.setState({sucessMessage: error})
      console.log(error)
    }
  }
  
  sendPhoto = async (file) => {
    let query = createPhoto('User',  this.props.id, 'User Photo')
    let photoOpts = {
      method: 'POST',
      headers: { "Content-Type": "application/json"},
      body: JSON.stringify({
        file: file
      })
    }
    try {
      await fetch(`https://staging-crowdhound-be.herokuapp.com/graphql?token=${this.props.token}&query=${query}`, photoOpts)
    } catch(error) {
      await console.log(error)
    }
  }

  render() {

    return (
      <section className="add-user-detail">
        <form className="add-user-detail-form">
          <h4 className="add-user-detail-title">Add your information</h4>
          <label for="user-detail-first-name-input" className="label">First Name</label>
          <p>{this.props.firstName}</p>
          <label for="user-detail-last-name-input" className="label">Last Name</label>
          <p>{this.props.lastName}</p>
          <label for="user-detail-email-input" className="label">Email</label>
          <p>{this.props.email}</p>
          <label for="user-detail-description-input" className="label">Description</label>
          <textarea id="user-detail-description-input" className="input"  type="text" value={this.state.description}
          name='description' onChange={this.handleChange}></textarea>
          <label for="user-detail-address-input" className="label">Address</label>
          <div className="address-container">
            <input className="address-input" type="text" placeholder="Street" value={this.state.street} name='street' onChange={this.handleChange}/>
            <input className="address-input" type="text" placeholder="City" value={this.state.city} name='city' onChange={this.handleChange}/>
            <input className="address-input" type="text" placeholder="State" value={this.state.st} name='st' onChange={this.handleChange}/>
            <input className="address-input" type="number"  max="5" placeholder="Zip" value={this.state.zip} name='zip' onChange={this.handleChange}/>
          </div>
          <label className="label">Photo</label>
          <ReactFileReader handleFiles={this.handleFiles} base64={true} multipleFiles={false}>
            <button id="user-detail-photo-input" className="input"  name='file' value='add-photo' type='button'>Add photo</button>
          </ReactFileReader>
          <Link to={`/userprofile/${this.props.id}`}>
            <input type="button" className="input" value="Add" id="add-user-detail-btn" onClick={this.sendUser}/> 
          </Link>
        </form>
      </section>
    )
  }
}

export const mapStateToProps = state => ({
  firstName: state.currentUser.firstName,
  lastName: state.currentUser.lastName,
  email: state.currentUser.email,
  token: state.currentUser.token,
  id: state.currentUser.id
})

export default connect(mapStateToProps)(AddUserDetail)
