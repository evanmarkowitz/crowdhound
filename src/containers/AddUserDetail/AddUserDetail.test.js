import { shallow } from 'enzyme';
import { AddUserDetail } from './AddUserDetail'
import React from 'react';


describe('AddUserDetail', () => {
  let wrapper;

  beforeEach(() => {

    wrapper = shallow(<AddUserDetail />)
  })
  
  it('should match the snapshot', () =>
    expect(wrapper).toMatchSnapshot()
  )


})