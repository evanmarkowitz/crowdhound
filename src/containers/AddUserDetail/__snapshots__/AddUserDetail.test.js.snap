// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddUserDetail should match the snapshot 1`] = `ShallowWrapper {}`;
